usaspending_api/data_load/fpds_loader.py

```diff
@@ -0,0 +1,260 @@
+import psycopg2
+import logging
+from os import environ
+
+from usaspending_api.data_load.field_mappings_fpds import (
+    transaction_fpds_columns,
+    transaction_normalized_columns,
+    transaction_normalized_functions,
+    legal_entity_columns,
+    legal_entity_boolean_columns,
+    legal_entity_functions,
+    recipient_location_columns,
+    recipient_location_functions,
+    place_of_performance_columns,
+    place_of_performance_functions,
+    award_functions,
+    transaction_fpds_boolean_columns,
+    transaction_fpds_functions,
+)
+from usaspending_api.data_load.data_load_helpers import capitalize_if_string, false_if_null
+from usaspending_api.data_load.generic_loaders import (
+    update_transaction_fpds,
+    update_transaction_normalized,
+    insert_transaction_normalized,
+    insert_transaction_fpds,
+    insert_recipient_locations,
+    insert_recipients,
+    insert_place_of_performance,
+    insert_award,
+)
+from usaspending_api.common.helpers.timing_helpers import Timer
+from usaspending_api.common.helpers.sql_helpers import get_database_dsn_string
+
+
+USASPENDING_CONNECTION_STRING = get_database_dsn_string()
+BROKER_CONNECTION_STRING = environ.get("DATA_BROKER_DATABASE_URL", None)
+
+DESTROY_ORPHANS_LEGAL_ENTITY_SQL = (
+    "DELETE FROM legal_entity legal WHERE legal.legal_entity_id in "
+    "(SELECT l.legal_entity_id FROM legal_entity l "
+    "LEFT JOIN transaction_normalized t ON t.recipient_id = l.legal_entity_id "
+    "LEFT JOIN awards a ON a.recipient_id = l.legal_entity_id "
+    "WHERE t is null and a.id is null); "
+)
+DESTROY_ORPHANS_REFERENCES_LOCATION_SQL = (
+    "DELETE FROM references_location location WHERE location.location_id in "
+    "(SELECT l.location_id FROM references_location l "
+    "LEFT JOIN transaction_normalized t ON t.place_of_performance_id = l.location_id "
+    "LEFT JOIN legal_entity e ON e.location_id = l.location_id "
+    "LEFT JOIN awards a ON a.place_of_performance_id = l.location_id "
+    "WHERE t.id is null and a.id is null and e.legal_entity_id is null)"
+)
+
+CHUNK_SIZE = 5000
```

Comment:
I guess I don't like that it's possible to call run_fpds_load on a list that is too big to actually load. If someone were to get a huge list in a clever, non-chunked way, the loader would then break for no reason.

Reply:
It's a fair point. Although the code itself won't break, it will just require larger amounts of memory than we think is prudent for routine operations.

If someone is using this function to load millions of records at once, it's their responsibility to handle the process resource needs. The "friendly" action to future developers would be to log warnings if the script receives more transactions than we recommend but execute anyway. However, just like the batching sizes we use, that warning is highly dependent on the data and the hardware running the code.